### PR TITLE
Added Automate Register Flow TestCase-4 

### PR DESCRIPTION
**Pull request by Dev-Bhagwani**

### CODE REVIEW

File: src/main/java/PageObjects/BookCart_HomePage.java [hunk 1]
```diff
@@ -14,6 +14,7 @@ public BookCart_HomePage() {
 
 	public static By bookCartLogo = By.xpath("//span[@class='mat-button-wrapper']");
 	public static By loginButton = By.xpath("//button[@class='mat-focus-indicator mat-button mat-button-base ng-star-inserted']");
+	public static By MatIconExistingUsername = By.xpath("(//span[@class='mat-button-wrapper'])[4]");
```
Comment (Sh0aibAli) [inline src/main/java/PageObjects/BookCart_HomePage.java:17]: use proper naming conventions.
Variable name should be start using a small letter case.

File: src/main/java/PageObjects/BookCart_HomePage.java [hunk 2]
```diff
@@ -23,4 +24,8 @@ public boolean DisplayingHomeCartLogo() {
 	public void clickingLoginButton() {
 		 driver.findElement(loginButton).click();
 	}
+	
+	public boolean  DisplayingMatIconExistingUserName() {
+		return driver.findElement(MatIconExistingUsername).isDisplayed();
```
Comment (Sh0aibAli) [inline src/main/java/PageObjects/BookCart_HomePage.java:29]: use proper naming conventions

File: src/main/java/PageObjects/BookCart_LoginPage.java [hunk 3]
```diff
@@ -12,15 +12,15 @@ public BookCart_LoginPage() {
 			PageFactory.initElements(driver, this);
 		}
 		
-		public static By loginHeader = By.xpath("//h3[text()='Login']");
+		public static By LoginHeader = By.xpath("//h3[text()='Login']");
```
Comment (PrekshaVora) [inline src/main/java/PageObjects/BookCart_LoginPage.java:15]: variable name should be used in a small letter case

File: src/main/java/PageObjects/BookCart_RegistrationPage.java [hunk 4]
```diff
@@ -22,6 +23,7 @@ public  BookCart_RegistrationPage() {
 	public static By EyeButton2 = By.xpath("(//div/mat-icon[@data-mat-icon-type='font'])[2]");
```
Comment (Sh0aibAli) [inline src/main/java/PageObjects/BookCart_RegistrationPage.java:23]: use proper naming conventions

File: src/main/java/PageObjects/BookCart_RegistrationPage.java [hunk 5]
```diff
@@ -63,4 +65,9 @@ public void ClickingRegisterButton() {
  		waitForElementToBeVisibile(RegisterButton);
 		driver.findElement(RegisterButton).click();
 	}
+ 	
+ 	public boolean DisplayingUserNameIsNotAvailable() {
```
Comment (Sh0aibAli) [inline src/main/java/PageObjects/BookCart_RegistrationPage.java:69]: Use proper naming conventions
Method name should be start by a small letter case

File: src/test/java/BookCart_TestPack/BookCart_TestClass.java [hunk 6]
```diff
@@ -34,7 +34,7 @@ public void BookCartRegister() throws IOException, InterruptedException{
 		registrationPage.ClickingEyeButton2();
 		registrationPage.ClickingGenderRadioButton();
 //		registrationPage.ClickingRegisterButton();
```
Comment (PrekshaVora) [inline src/test/java/BookCart_TestPack/BookCart_TestClass.java:36]: Remove commented code

File: src/test/java/BookCart_TestPack/BookCart_TestClass.java [hunk 7]
```diff
@@ -34,7 +34,7 @@ public void BookCartRegister() throws IOException, InterruptedException{
...
-		
+		Thread.sleep(3000);
```
Comment (Sh0aibAli) [inline src/test/java/BookCart_TestPack/BookCart_TestClass.java:37]: Remove thread.sleep and use selenium waits instead.

**Review by Sh0aibAli — CHANGES_REQUESTED** [review]
Please check comments

**Review by PrekshaVora — CHANGES_REQUESTED** [review]

**Pull request merged**